code/model/model.h

```diff
@@ -482,12 +482,28 @@ struct w_bank
 	vec3d	*norm = nullptr;
 	float   *external_model_angle_offset = nullptr;
 
+	w_bank() { }
+
 	~w_bank()
 	{
 		delete[] pnt;
 		delete[] norm;
 		delete[] external_model_angle_offset;
 	}
+
+	w_bank& operator=(w_bank&& other) {
+		this->~w_bank();
+		num_slots = other.num_slots;
+		pnt = other.pnt;
+		norm = other.norm;
+		external_model_angle_offset = other.external_model_angle_offset;
+		other.pnt = nullptr;
+		other.norm = nullptr;
+		other.external_model_angle_offset = nullptr;
+		return *this;
+	}
+	w_bank(const w_bank& other) = default;
+	w_bank& operator=(const w_bank& other) = default;
```

Comment:
It doesn't.
But that also _shouldn't_ be a problem.
If you copy the whole structure, both copies point to the same heap info.
What _is_ a potential problem, is if one gets destructed, the other one's arrays are also deleted.
But since the code here will immediately reassign the pointers after copying, this is avoided.
The original thus retains its pointer to the heap and deallocates it on its destruction, while the copy gets its own copy allocated later on.

Reply:
It seems risky to rely on usage for correct behavior, when the correct behavior should really be enforced by the class definition.  What if a variable contained a `w_bank`, then was assigned another?  The heap data would be lost.  If that scenario isn't present in the current code, there's no guarantee it won't be present in future code.  And I didn't even think of your other potential problem, but the same risk applies there too.

I would say that either the class definition should enforce the correct behavior, or at the very least a comment should be added explaining the risks and how they are avoided.

Reply:
> What if a variable contained a w_bank, then was assigned another?
Right, this is the one case in which we could leak memory.
To alleviate this, I just delete the copy assignment while leaving the copy constructor.

Reply:
All right.